### PR TITLE
Allow symfony 6

### PR DESCRIPTION
**Pull request by akalineskou**
These were the changes I've made to allow symfony 6 to work.

The biggest one was `query->all` since `get` checks that the value is scalar and it was throwing an error since it was an array.
The other ones were deprecations for the return types.

### CODE REVIEW

File: Query/ConditionManager.php [hunk 1]
```diff
@@ -54,27 +54,27 @@ public function offsetSet($offset, $value)
...
 
-    public function key() {
+    public function key(): mixed {
         return key($this->conditions);
     }
 
-    public function next() {
-        return next($this->conditions);
+    public function next(): void {
+        next($this->conditions);
```
Comment (denisvmedia) [inline Query/ConditionManager.php:74]: What is the reason of this change? [Docs](https://www.php.net/manual/en/function.next.php) say:

> Returns the array value in the next place that's pointed to by the internal array pointer, or false if there are no more elements.

File: composer.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
   ],
   "require": {
     "php": ">=7.3.0",
```
Comment (denisvmedia) [inline composer.json]: You should then also increase php version to `>=8.0.0`. Because 7.3.* versions don't support typed properties, which you used in your PR. And php 7.4.* versions don't `mixed` type.

Comment (akalineskou) [reply composer.json]: Done

**Comment by denisvmedia** [comment]
Note, as PHP 7.3 and 7.4 versions are now removed, we need to disable the tests for them.
We also need the tests for Symfony 6.0.

P.S. The code is fine, I will merge it, as soon as we have the tests green.

**Comment by akalineskou** [comment]
Done.

Updated symfony version 5 to 5.4 since the other ones are not supported anymore...
Also added php 8.1 for future releases (let me know if you dont want it tested)

**Comment by akalineskou** [comment]
Fixed the tests, also added try catch to the `get` in case of a `BadRequestException` to return the `InvalidArgumentException`

File: Request/Request.php [hunk 3]
```diff
@@ -49,18 +50,42 @@ final class Request
      */
     public function __construct(HttpRequest $request)
     {
-        $this->pageNum = (int)$request->query->get('page', 1);
-        $this->limit = (int)$request->query->get('limit', -1);
-        $this->query = $request->query->get('filter');
-        if ($this->query !== null && !is_array($this->query)) {
+        try {
+            $this->pageNum = (int)$request->query->get('page', 1);
+        } catch (BadRequestException) {
```
Comment (denisvmedia) [inline Request/Request.php:55]: Hm... Are you sure `$request->query->get` can return BadRequestException?

Comment (akalineskou) [reply Request/Request.php:55]: It does throw one https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/InputBag.php#L37

The tests should also prove this since it returns the correct exception

Comment (denisvmedia) [reply Request/Request.php:55]: Aha, ok. Thank you for this change.

**Review by denisvmedia — APPROVED** [review]
LGTM. Thank you for your efforts. Merging.

**Comment by akalineskou** [comment]
Hey, when can we expect a new release/tag?

**Comment by denisvmedia** [comment]
I will tag it tonight.
